### PR TITLE
New TabbedPanel (improves mobile support) & shift-drag zoom box

### PR DESCRIPTION
**Pull request by glenselle**
Resolves: #149 #148 also includes changes from: https://github.com/MonsantoCo/ol-kit/pull/240

### PR Safety Checklist:

 - [x] Added the task to the appropriate release doc under **Enhancements** or **Bug Fixes**
 - [x] Bump `package.json` & `package-lock.json` version numbers to appropriate release
 - [x] (optional) All external API changes have been documented
 - [x] (optional) Build docs: `npm run docs`

### CODE REVIEW

File: app/demos/inline/App.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    const layer = new VectorTileLayer({
+      source: new VectorTileSource({
+        format: new MVT(),
+        url: `http://localhost:7800/public.gis_osm_waterways_07_1/{z}/{x}/{y}.pbf`,
+        // url: `https://free-{1-3}.tilehosting.com/data/v3/{z}/{x}/{y}.pbf.pict`,
+        maxZoom: 14
+      })
+    });
+    map.addLayer(layer)
```
Comment (glenselle) [inline app/demos/inline/App.js:32]: remove this

File: app/Router.js [hunk 2]
```diff
@@ -5,21 +5,16 @@ import {
...
 import Covid from './demos/covid/App'
 import Space from './demos/space/App'
+import Inline from './demos/inline/App'
 import World from './demos/world/App'
 import Geohack from './demos/geohack/App'
 
 export default function App () {
   return (
     <Router>
-      <Snackbar />
```
Comment (glenselle) [inline app/Router.js:20]: Add link to source code GH

File: src/Controls/utils.js [hunk 3]
```diff
@@ -1,5 +1,20 @@
 import { easeOut, linear } from 'ol/easing'
 
+const OLZOOMBOXID = 'ol-css-zoombox-style'
```
Comment (glenselle) [inline src/Controls/utils.js:3]: put comment on this

File: src/Controls/utils.js [hunk 4]
```diff
@@ -1,5 +1,20 @@
...
+
+export function replaceZoomBoxCSS () {
```
Comment (glenselle) [inline src/Controls/utils.js:5]: call this function globally in the `Map`

File: src/Controls/ZoomOut.js [hunk 5]
```diff
@@ -39,6 +39,8 @@ function ZoomOut (props) {
     clearTimeout(repeatTimeout)
   }
 
+  if (injectZoomBoxCSS) replaceZoomBoxCSS()
```
Comment (glenselle) [inline src/Controls/ZoomOut.js:42]: remove these

File: src/Controls/ZoomIn.js [hunk 6]
```diff
@@ -42,6 +42,8 @@ function ZoomIn (props) {
     clearTimeout(repeatTimeout)
   }
 
+  if (injectZoomBoxCSS) replaceZoomBoxCSS()
```
Comment (glenselle) [inline src/Controls/ZoomIn.js:45]: remove this

File: src/LayerPanel/LayerPanelActionImport/utils.js [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+        })
+      })
+      .catch(reject)
+  })
+}
+
+/**
+ * Converts the given file into an array of features
+ * @function convertFileToFeatures
+ * @since 7.0.0
```
Comment (glenselle) [inline src/LayerPanel/LayerPanelActionImport/utils.js:50]: change these

File: src/LayerPanel/LayerPanelLayersPage/index.js [hunk 8]
```diff
@@ -268,16 +270,39 @@ class LayerPanelLayersPage extends Component {
     this.setState({ layers: reorderedLayers })
   }
 
+  onFileImport = file => {
+    const { map, onFileImport } = this.props
+console.log('handling import?????')
```
Comment (glenselle) [inline src/LayerPanel/LayerPanelLayersPage/index.js:275]: remove

File: src/LayerPanel/LayerPanelLayersPage/index.js [hunk 9]
```diff
@@ -268,16 +270,39 @@ class LayerPanelLayersPage extends Component {
...
+    // otherwise, add them to the map ourselves
+    convertFileToFeatures(file, map).then(({ features, name }) => {
+      console.log('should have converted...')
```
Comment (glenselle) [inline src/LayerPanel/LayerPanelLayersPage/index.js:278]: remove

File: src/LayerStyler/StyleManager/index.js [hunk 10]
```diff
@@ -167,52 +167,41 @@ class StyleManager extends Component {
...
+                  }}>
+                  {layerTitles.map(t => {
+                    return <MenuItem key={t} value={t}>{t}</MenuItem>
+                  })}
+                </Select>
+              </FormControl>
+            </InputContainer>
+          </HeaderContainer>
+        ) : (
+          <NoLayerText>You must have a layer on the map to style</NoLayerText>
```
Comment (glenselle) [inline src/LayerStyler/StyleManager/index.js:202]: translations

File: src/LayerStyler/StyleManager/index.js [hunk 11]
```diff
@@ -259,6 +248,10 @@ class StyleManager extends Component {
   }
 }
 
+StyleManager.defaultProps = {
+  initialIdx: 0
```
Comment (glenselle) [inline src/LayerStyler/StyleManager/index.js:252]: revert this

File: src/MapPanel/MapPanel.js [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+import KeyboardArrowRightIcon from '@material-ui/icons/KeyboardArrowRight'
+import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown'
+import { connectToContext } from 'Provider'
+
+/**
+ * @component
+ * @category MapPanel
+ * @since 1.3.0
+ */
+class MapPanel extends Component {
```
Comment (glenselle) [inline src/MapPanel/MapPanel.js:14]: TabbedPanel

File: src/MapPanel/MapPanel.js [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+            <Tabs
+              open={showPanel}
+              value={activeIndex}
+              onChange={this.handleChange}
+              variant='scrollable'
+              scrollButtons='auto'>
+              {React.Children.map(children, (child, i) => {
+                if (child) return <Tab key={i + 1} label={child.props.tabIcon || child.props.label} />
+              })
+              }
```
Comment (glenselle) [inline src/MapPanel/MapPanel.js:54]: move up

File: src/MapPanel/MapPanel.js [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+MapPanel.defaultProps = {}
+
+MapPanel.propTypes = {
+  /** The pages of the panel (things like `LayerPanelPage` or `LayerStyler` components) */
+  children: PropTypes.node.isRequired,
+
+  /** An object of styles spread on the map panel */
+  style: PropTypes.object,
```
Comment (glenselle) [inline src/MapPanel/MapPanel.js:71]: drop til need arises

File: src/MapPanel/MapPanel.js [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+
+  /** A float number for the opacity of the LayerPanel. i.e. (0.5) */
+  opacity: PropTypes.number,
```
Comment (glenselle) [inline src/MapPanel/MapPanel.js:74]: pass through

File: src/MapPanel/styled.js [hunk 16]
```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import MaterialCard from '@material-ui/core/Card'
+import MaterialTabs from '@material-ui/core/Tabs'
+import MaterialTab from '@material-ui/core/Tab'
+import MaterialCheckbox from '@material-ui/core/Checkbox'
+import MaterialCardContent from '@material-ui/core/CardContent'
+import { styled } from '@material-ui/styles'
+
+export const Card = styled(({ ...props }) => <MaterialCard {...props} />)({
+  position: props => props.inline ? 'inline' : 'absolute',
```
Comment (glenselle) [inline src/MapPanel/styled.js:10]: inline is not a thing for this card...always absolute b/c we're positioned absolute to the map container

File: src/PlaceSearch/PlaceSearch.js [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+import { SearchContainer, PlaceSearchInput } from './styled'
+
+import { connectToContext } from 'Provider'
+
+/**
+ * @component
+ * @category PlaceSearch
+ * @since 1.3.0
+ */
+class PlaceSearch extends Component {
```
Comment (glenselle) [inline src/PlaceSearch/PlaceSearch.js:12]: revert this

File: src/PlaceSearch/styled.js [hunk 18]
```diff
@@ -0,0 +1,16 @@
+import styled from 'styled-components'
+
+export const PlaceSearchInput = styled.input`
```
Comment (glenselle) [inline src/PlaceSearch/styled.js:3]: revert this

File: app/Router.js (hunk 2 above)
Comment (glenselle) [reply app/Router.js:20]: ✅

**Comment by drodenberg** [comment]
We should really work on getting this PR merged

**Comment by stazrad** [comment]
> We should really work on getting this PR merged

what is it waiting on? @glenselle @drodenberg?

**Comment by stazrad** [comment]
Docs are built on this branch as well as demo site improvements and bug fixes. This needs to go out with `1.4.0` but some tests are failing @drodenberg 👀

**Review by drodenberg — APPROVED** [review]

**Pull request merged**